contracts/modules/TransferManager/SingleTradeVolumeRestrictionManager.sol

```diff
@@ -0,0 +1,176 @@
+        validTransfer = (_amount.mul(10**18).div(ISecurityToken(securityToken).totalSupply())) <= globalTransferLimitInPercentage;
+      } else {
+        validTransfer = _amount <= globalTransferLimit;
+      }
+    }
+    if(validTransfer) return Result.NA;
+    return Result.INVALID;
+  }
+
+  /**
+  * @notice Used to intialize the variables of the contract
+  * @param _isTransferLimitInPercentage true if the transfer limit is in percentage else false
+  * @param _globalTransferLimitInPercentageOrToken transfer limit per single transaction.
+  */
+  function configure(bool _isTransferLimitInPercentage, uint256 _globalTransferLimitInPercentageOrToken) public onlyFactory {
+    require(_globalTransferLimitInPercentageOrToken > 0, "global transfer limit has to greater than 0");
+    isTransferLimitInPercentage = _isTransferLimitInPercentage;
+    if(isTransferLimitInPercentage) {
+      globalTransferLimitInPercentage = _globalTransferLimitInPercentageOrToken;
+    } else {
+      globalTransferLimit = _globalTransferLimitInPercentageOrToken;
+    }
+  }
+
+  /**
+  * @notice Change the global transfer limit
+  * @param _newGlobalTransferLimit new transfer limit in tokens
+  */
+  function changeGlobalLimitInTokens(uint256 _newGlobalTransferLimit) public withPerm(ADMIN) {
+    require(!isTransferLimitInPercentage, "Transfer limit not set in tokens");
+    globalTransferLimit = _newGlobalTransferLimit;
+    emit LogGlobalTransferLimitInTokensSet(_newGlobalTransferLimit);
+  }
+
+  /**
+  * @notice Change the global transfer limit
+  * @param _newGlobalTransferLimitInPercentage new transfer limit in percentage. Multiple the percentage by 10^16. Eg 22% will be 22*10^16
+  */
+  function changeGlobalLimitInPercentage(uint256 _newGlobalTransferLimitInPercentage) public withPerm(ADMIN) {
+    require(isTransferLimitInPercentage, "Transfer limit not set in Percentage");
+    globalTransferLimitInPercentage = _newGlobalTransferLimitInPercentage;
+    emit LogGlobalTransferLimitInPercentageSet(_newGlobalTransferLimitInPercentage);
+  }
+
+  /**
+  * @notice add an exempt wallet
+  * @param _wallet exempt wallet address
+  */
+  function addExemptWallet(address _wallet) public withPerm(ADMIN) {
+    require(_wallet != address(0), "Wallet address cannot be a zero address");
+    exemptWallets[_wallet] = true;
+    emit LogExemptWalletAdded(_wallet);
+  }
+
+  /**
+  * @notice remove an exempt wallet
+  * @param _wallet exempt wallet address
+  */
+  function removeExemptWallet(address _wallet) public withPerm(ADMIN) {
+    require(_wallet != address(0), "Wallet address cannot be a zero address");
+    exemptWallets[_wallet] = false;
+    emit LogExemptWalletRemoved(_wallet);
+  }
+
+  /**
+  * @notice set transfer limit per wallet an exempt wallet
+  * @param _wallet wallet address
+  * @param _transferLimit transfer limit for the wallet in tokens
+  */
+  function setTransferLimitForWallet(address _wallet, uint _transferLimit) public withPerm(ADMIN) {
```

Comment:
Would be ideal to make this into a batch function. Pass _wallet and _transferLimit as arrays and perform this operation for each item. So on the frontend we can perform these operations in batch.